nexus/src/internal_api/http_entrypoints.rs

```diff
@@ -484,3 +488,78 @@ async fn saga_view(
     };
     apictx.external_latencies.instrument_dropshot_handler(&rqctx, handler).await
 }
+
+// NAT RPW internal APIs
+
+/// Path parameters for NAT ChangeSet
+#[derive(Deserialize, JsonSchema)]
+struct RpwNatPathParam {
+    /// which change number to start generating
+    /// the change set from
+    from_gen: u32,
+}
+
+/// Query parameters for NAT ChangeSet
+#[derive(Deserialize, JsonSchema)]
+struct RpwNatQueryParam {
+    limit: u32,
+}
+
+/// Fetch NAT ChangeSet
+///
+/// Caller provides their generation as `from_gen`, along with a query
+/// parameter for the page size (`limit`). Endpoint will return changes
+/// that have occured since the caller's generation number up to the latest
+/// change or until the `limit` is reached. If there are no changes, an
+/// empty vec is returned.
+#[endpoint {
+   method = GET,
+    path = "/rpw/nat/ipv4/changeset/{from_gen}"
```

Comment:
tiny nitpick -- I'd just prefix these paths with "/nat", the fact that they're called from Dendrite is an implementation detail

Reply:
I think I still might remove the `/rpw` prefix, the fact that Nexus processes this as an RPW is an implementation detail.